### ISSUE TITLE
Portugal States

### ISSUE THREAD

**Issue by lcestaro** [woocommerce/woocommerce#29206]
https://github.com/woocommerce/woocommerce/blob/35d4ee4073a180d6fa07fa63a4fce37c394cfa1c/i18n/states.php#L1299

	'PT' => array( // Portugal states.
		'PT-01'   => __( 'Aveiro', 'woocommerce' ),
		'PT-02'   => __( 'Beja', 'woocommerce' ),
		'PT-03'   => __( 'Braga', 'woocommerce' ),
		'PT-04'   => __( 'Bragança', 'woocommerce' ),
		'PT-05'   => __( 'Castelo Branco', 'woocommerce' ),
		'PT-06'   => __( 'Coimbra', 'woocommerce' ),
		'PT-07'   => __( 'Évora', 'woocommerce' ),
		'PT-08'   => __( 'Faro', 'woocommerce' ),
		'PT-09'   => __( 'Guarda', 'woocommerce' ),
		'PT-10'  => __( 'Leiria', 'woocommerce' ),
		'PT-11'  => __( 'Lisboa', 'woocommerce' ),
		'PT-12'  => __( 'Portalegre', 'woocommerce' ),
		'PT-13'  => __( 'Porto', 'woocommerce' ),
		'PT-14'  => __( 'Santarém', 'woocommerce' ),
		'PT-15'  => __( 'Setúbal', 'woocommerce' ),
		'PT-16'  => __( 'Viana do Castelo', 'woocommerce' ),
		'PT-17'  => __( 'Vila Real', 'woocommerce' ),
		'PT-18'  => __( 'Viseu', 'woocommerce' ),
		'PT-20'  => __( 'Açores', 'woocommerce' ),
		'PT-30'  => __( 'Madeira', 'woocommerce' ),
	),

Reference: https://en.wikipedia.org/wiki/ISO_3166-2:PT

**Comment by juliaamosova** [comment]
Hi @lcestaro,

Thank you for opening the thread. Can you please confirm what do you like to be done regarding the Portugal States? Are you looking for the states to be added to WooCommerce? It looks like this is what you are looking for but it is always best to confirm to avoid misunderstanding.

**Comment by lcestaro** [comment]
Hello @juliaamosova,

Yes, I am looking for states to use district taxes. I believe that these settings can be useful, just as it is for me. It could be integrated into Woocommerce.

**Comment by masteradhoc** [comment]
@lcestaro added a PR for this, thanks!

**Comment by webdados** [comment]
As much as the inclusion of Portuguese states (known as Distritos) are welcomed, there's something I think should be well thought before releasing this to production:

- There's already a very well established plugin that adds the Portuguese Distrito to WooCommerce: https://pt.wordpress.org/plugins/portugal-states-distritos-for-woocommerce/ (disclaimer: I'm the author) and the addition of the districts to the WooCommerce core should at least mimic the same array keys, or else this release has the potential to break a lot of websites if the 4000+ Portuguese shops that use this plugin decide to deactivate it because the core already has the districts on it: https://plugins.trac.wordpress.org/browser/portugal-states-distritos-for-woocommerce/trunk/woocommerce-portugal-states.php#L69 (all older order data and current shipping zones configurations will break because the keys are not the same)

- AFAIK, the reason WooCommerce didn't include the Portuguese districts it's because they're not textually part of the Portuguese address as can be seen at https://github.com/woocommerce/woocommerce/blob/trunk/includes/class-wc-countries.php#L540 , so IMHO I think "Portuguese states" is plugin territory and not core

If you are completely decided to go ahead with this, at least please mimic the same array keys for the states or a LOT of stores (and all its old orders data) are on the line to be broken.

**Comment by webdados** [comment]
> Yes, I am looking for states to use district taxes.

Although Portugal has different taxes in Madeira and Azores, the tax should be charged based on the store address (State = District) and not based on the customer address, so it makes no sense to set taxes based on the district because if the store is based on the mainland you will charge that tax to mainland, Madeira or Azores customers. The same happens if the store is on the islands and you're selling to the mainland: the customer will pay the mainland taxes.
In short: Portuguese states are never used for tax calculation.

The only reason the "states" should exist on the WooCommerce core is when the state name itself is used on the address textual format, which doesn't happen in Portugal.
